packaging/leapp-supplements.spec

```diff
@@ -0,0 +1,111 @@
+# Define Globals
+%global leapp_datadir %{_datadir}/leapp-repository
+%global repositorydir %{leapp_datadir}/repositories
+%global custom_repositorydir %{leapp_datadir}/custom-repositories
+%global supplementsdir %{repositorydir}/system_upgrade_supplements
+
+%if 0%{?rhel} == 7
+    %define rpm_name leapp-supplements-el7toel8
+%endif
+%if 0%{?rhel} == 8
+    %define rpm_name leapp-supplements-el8toel9
+%endif
+
+# Define RPM Preamble
+Name:           %{rpm_name}
+Version:        1.0.0
+Release:        1%{?dist}
+Summary:        Custom actors for the Leapp project
+
+License:        MIT
+URL:            https://github.com/oamg/%{name}
+Source0:        %{name}-%{version}.tar.gz
+
+BuildArch:      noarch
+
+%if 0%{?rhel} == 7
+### RHEL 7 ###
+BuildRequires:  python-devel
+
+Requires:       leapp
+Requires:       python2-leapp
+# Dependency on leapp-repository
+Requires:       leapp-upgrade-el7toel8
+%endif
+
+%if 0%{?rhel} == 8
+### RHEL 8 ###
+BuildRequires:  python3-devel
+
+Requires:       leapp
+Requires:       python3-leapp
+# Dependency on leapp-repository
+Requires:       leapp-upgrade-el8toel9
+%endif
+
+%description
+Custom leapp actors for the in-place upgrade to the next major version
+of the Red Hat Enterprise Linux system.
+
+%prep
+%setup -q
+
+%install
+install -m 0755 -d %{buildroot}%{custom_repositorydir}
+install -m 0755 -d %{buildroot}%{repositorydir}
+install -m 0755 -d %{buildroot}%{supplementsdir}
+install -m 0755 -d %{buildroot}%{_sysconfdir}/leapp/repos.d/
+cp -r repos/system_upgrade_supplements/* %{buildroot}%{supplementsdir}
+
+# Remove actors not found in actors_to_install list
+for subdir in common el7toel8 el8toel9; do \
+    [ -d %{buildroot}%{supplementsdir}/$subdir/actors ] || continue; \
+    pushd %{buildroot}%{supplementsdir}/$subdir/actors; \
+        for actor in *; do \
+            # Check if actor is in actors_to_instal llist
+            if ! echo "%{actors_to_install}" | grep -qw "$actor"; then \
+                rm -rf $actor; \
+            fi; \
+        done; \
+    popd; \
+done
```

Comment:
This will work only if actors will be put directly to the actors directory only. If any actors are put in additional subdirs (e.g. to have all apache actors together), such actors will not be discovered. Is discussed on the mtg, it depends what rules are set for people contributing to this repository. But if the ability to have actors stored in other subdirs is expected, I suggest to search actors' directories based on actor.py file instead of going in this way. 

E.g. something like (untested, writting in the train, sorry):
```
find %{buildroot}%{supplementsdir}/ -name actor.py -exec dirname {} \;  > tmp_actor_paths
while read -r actor_path; do
  actor_name=$(basename "$actor_path")
  echo "%{actors_to_install}" | grep -qw "${actor_name}" || rm -rf "$actor_path"  
done < tmp_actor_paths

# remove remaining empty dirs
find %{buildroot}%{supplementsdir}/ -type d -depth -empty -delete
```

Also to make it "faster", I would probably first remove irrelevant repositories (below) and than I would proceed with this additional removal of unwanted actors

Reply:
Nice, I did not think about nested subdirs. Thanks for the code example!